### DISCUSSION TITLE
Connect LiDAR to ESP32-S3

### DISCUSSION THREAD

**Discussion by makerspet** [makerspet/support#4] [Support Q&A] [answered]
Which TX, RX pins to use when connecting a LiDAR to ESP-S3?

**Comment by kaiaai** [comment, marked as answer]
One solution can be to use ESP32-S3 hardware serial port 1 and its default TX, RX pins:
- use GPIO17 TXD1
- use GPIO18 RXD1
- modify the Arduino [LiDAR library example](https://github.com/kaiaai/LDS/blob/main/examples/lds_basic_esp32/lds_basic_esp32.ino) to use hardware serial port 1
```
HardwareSerial LidarSerial(1);
```
